lib/eventboss/unit_of_work.rb

```diff
@@ -12,12 +12,27 @@ def initialize(client, queue, listener, message)
       @listener = listener
       @message = message
       @logger = logger
+
+      listener_instance = @listener.new
+      if listener_instance.respond_to?(:receive)
+        receive_method_params = listener_instance.method(:receive).parameters
+        if [:key, :keyreq].include?(receive_method_params[0][0])
+          @listener.required_params = receive_method_params.filter { |p| p[0] == :keyreq }.map { |p| p[1] }
+          @listener.optional_params = receive_method_params.filter { |p| p[0] == :key }.map { |p| p[1] }
+        end
+      end
     end
 
     def run
       logger.debug(@message.message_id) { 'Started' }
       processor = @listener.new
-      processor.receive(JSON.parse(@message.body))
+      JSON.parse(@message.body).tap do |payload|
```

Comment:
```suggestion
      JSON.parse(@message.body).then do |payload|
```
the effect is the same, but `then` seems more semantic to me in this case.